docs/examples/cert_revocation_check.c

```diff
@@ -0,0 +1,176 @@
+      for(int k = 0; k < sk_GENERAL_NAME_num(distpoint->name.fullname); k++) {
+        GENERAL_NAME *gen;
+        ASN1_IA5STRING *asn1_str;
+
+        gen = sk_GENERAL_NAME_value(distpoint->name.fullname, k);
+        asn1_str = gen->d.uniformResourceIdentifier;
+        *crlUrl = (char *)ASN1_STRING_data(asn1_str);
+
+        return 1;
+      }
+    }
+    else if(distpoint->type == 1) { /* relativename X509NAME */
+      STACK_OF(X509_NAME_ENTRY) *sk_relname = distpoint->name.relativename;
+      for(int k = 0; k < sk_X509_NAME_ENTRY_num(sk_relname); k++) {
+        X509_NAME_ENTRY *e;
+        ASN1_STRING *d;
+
+        e = sk_X509_NAME_ENTRY_value(sk_relname, k);
+        d = X509_NAME_ENTRY_get_data(e);
+
+        *crlUrl = (char *)ASN1_STRING_data(d);
+        return 1;
+      }
+    }
+  }
+
+  return 0;
+}
+
+static CURLcode user_verify_status_callback(CURL *curl, void *s, void *userptr)
+{
+  SSL *ssl = NULL;
+  int chainLength = 0;
+  STACK_OF(X509) *ch = NULL;
+  unsigned char *status = NULL;
+  CURLcode result = CURLE_OK;
+  char nameBuffer[256];
+  long len;
+
+  (void)curl; /* unused */
+  (void)userptr; /* unused */
+
+  fprintf(stdout, "user_verify_status_callback\n");
+
+  ssl = (SSL *)s;
+  len = SSL_get_tlsext_status_ocsp_resp(ssl, &status);
+
+  /* ocsp stappling was not received, lets try to download crl and check
+   * certificate status using CRL */
+  if(!status) {
+    fprintf(stdout,
+            "SSL_get_tlsext_status_ocsp_resp: ocsp status was not received\n");
+
+    ch = SSL_get_peer_cert_chain(ssl);
+
+    if(!ch) {
+      fprintf(stderr, "SSL_get_peer_cert_chain failed\n");
+      result = CURLE_SSL_INVALIDCERTSTATUS;
+      goto end;
+    }
+
+    chainLength = sk_X509_num(ch);
+
+    for(int i = 0; i < chainLength; ++i) {
+      const char *crlUrl = NULL;
+      const char *ocspUrl = NULL;
+      X509 *certificate = sk_X509_value(ch, chainLength - i - 1);
+
+      fprintf(stdout, "Certificate info %i:\n", i);
+
+      fprintf(stdout,
+              "\tsubj:   `%s`\n",
+              X509_NAME_oneline(X509_get_subject_name(certificate),
+                                nameBuffer,
+                                sizeof(nameBuffer)));
+
+      fprintf(stdout,
+              "\tissuer: `%s`\n",
+              X509_NAME_oneline(X509_get_issuer_name(certificate),
+                                nameBuffer,
+                                sizeof(nameBuffer)));
+
+      if(get_crl_url(certificate, &crlUrl)) {
+        fprintf(stdout, "\tCRL url: `%s`\n", crlUrl);
+
+        /* TODO: download CRL, verify its signature and check certificate for
+         * revocation */
+      }
+    }
+
+    goto end;
+  }
+
+  /* TODO: do some oscp status verification checks here */
+
+end:
+  return result;
+}
+
+int main(void)
+{
+  CURL *curl;
+  CURLcode res;
+
+  curl_global_init(CURL_GLOBAL_DEFAULT);
+
+  curl = curl_easy_init();
+  if(curl) {
+    curl_easy_setopt(curl, CURLOPT_URL, "https://some-site.com");
```

Comment:
I propose using `https://example.com` instead, as that's made for this sort of thing!

Reply:
it was fixed in the second commit as curl_easy_setopt(curl, CURLOPT_URL, argv[1]);